modules/artifact/src/main/java/org/mule/runtime/module/artifact/classloader/MuleArtifactClassLoader.java

```diff
@@ -110,20 +124,42 @@ public void setResourceReleaserClassLocation(String resourceReleaserClassLocatio
   }
 
   protected ResourceReleaser createResourceReleaserInstance() {
+    return (ResourceReleaser) createCustomInstance(resourceReleaserClassLocation);
+  }
+
+  private Object createCustomInstance(String classLocation) {
     InputStream classStream = null;
     try {
-      classStream = this.getClass().getResourceAsStream(resourceReleaserClassLocation);
+      classStream = this.getClass().getResourceAsStream(classLocation);
       byte[] classBytes = IOUtils.toByteArray(classStream);
       classStream.close();
       Class clazz = this.defineClass(null, classBytes, 0, classBytes.length);
-      return (ResourceReleaser) clazz.newInstance();
+      return clazz.newInstance();
     } catch (Exception e) {
-      throw new RuntimeException("Can not create resource releaser instance from resource: " + resourceReleaserClassLocation, e);
+      throw new RuntimeException("Can not create instance from resource: " + classLocation, e);
     } finally {
       closeQuietly(classStream);
     }
   }
 
+  /**
+   * Setup reactor-core error hooks, these are required for plugins that end up executing reactive streams.
+   *
+   * For instance, compatibility plugin which inherits from transformers that call {@code block()}.
+   */
+  protected void configureErrorHooks() {
+    try {
+      Class reactorHooks = loadClass("reactor.core.publisher.Hooks");
+      if (reactorHooks.getClassLoader().equals(this)) {
+        createCustomInstance("/org/mule/module/artifact/classloader/ErrorHooksConfiguration.class");
+      }
+    } catch (ClassNotFoundException e) {
+      // ignore, we don't care if the plugin does not include reactor-core
+    } catch (Exception e) {
+      logger.error("Cannot configure error hooks", e);
```

Comment:
If you are referring to https://github.com/mulesoft/mule/pull/5390#discussion_r130721565, I didn't mean to ignore